### PR TITLE
Updates on Room Search Issue #20 & Issue#21

### PR DESCRIPTION
**Pull request by ehharvey**
Merged Microservice of Room Search & Room Management

### CODE REVIEW

File: api/service_routes/room-management.yaml [hunk 1]
```diff
@@ -3,18 +3,139 @@ info:
...
+          description: The numbers of items to return
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                properties:
+                  "Student":
+                    $ref: "#/components/schemas/Student"
```
Comment (ehharvey) [inline api/service_routes/room-management.yaml:137]: Should be a list of rooms

File: api/service_routes/room-management.yaml [hunk 2]
```diff
@@ -3,18 +3,139 @@ info:
   title: API Title
   version: '1.0'
 servers:
-  - url: https://room-management.reserva.emilharvey.ca
+  - url: https://api.reserva.emilharvey.ca
```
Comment (ehharvey) [inline api/service_routes/room-management.yaml:6]: Change to room.reserva...

File: api/service_routes/room-management.yaml [hunk 3]
```diff
@@ -3,18 +3,154 @@ info:
...
+    put:
+      tags: 
+        - Room Management
+      summary: Updates a Room object
+      description: >
+        ## More Information
+
+        Request for `PUT /rooms/{id}` requires a body that contains all information of the Room except for its id. 
+        (The id of the room will be auto-incremented).
+        Response body shows the body that contains
```
Comment (ehharvey) [inline api/service_routes/room-management.yaml:45]: The `id` won't be auto incremented. `put` here should be for updating an existing room.


```suggestion
        Updates an existing room (identified by `id`).
```

**Comment by ehharvey** [comment]
@Amanuel-Negussie looking good. See the above for some more notes.

One thing I noticed is that your descriptions for routes is long, which is nice, but it probably not needed. See [this](https://github.com/ehharvey/Reserva/pull/37#discussion_r1146989316) for an example description which I think is fine (maybe a little short, lmk what you think)

**Review by Amanuel-Negussie — APPROVED** [review]

File: api/service_routes/room-management.yaml [hunk 4]
```diff
@@ -3,18 +3,150 @@ info:
...
+      description: >
+        ## More Information
+
+        Request for `GET /rooms/{id}` requires an id which will be in the search parameter.
+        Response will be in JSON format with S
+      operationId: getRoomId
+      parameters:
+        - name: id
+          in: path
+          description: User ID
```
Comment (ehharvey) [inline api/service_routes/room-management.yaml:22]:
```suggestion
          description: Room ID
```

File: api/schemas/Room.yaml [hunk 5]
```diff
@@ -4,7 +4,16 @@ properties:
...
     type: string
-    example: Large Room with Low Ceilings
\ No newline at end of file
+    example: Large Room with Low Ceilings
+  features:
+    type: array
+    items: 
+      anyOf:
+        - $ref: "./BooleanFeature.yaml"
+        - $ref: "./NumericalFeature.yaml"
```
Comment (ehharvey) [inline api/schemas/Room.yaml:17]: We should define a schema for `Feature`, `BooleanFeature`, and `NumericalFeature`

**Review by Dom17Pham — APPROVED** [review]

**Pull request merged**